client/validator.go

```diff
@@ -52,25 +53,28 @@ var (
 )
 
 type Validator struct {
-	c    *http.Client
-	conf *config.Provider
+	c              *http.Client
+	contextualizer contextx.Contextualizer
+	defaultConf    *config.Provider
 }
 
-func NewValidator(conf *config.Provider) *Validator {
+func NewValidator(contextualizerProvider contextx.Contextualizer, defaultConf *config.Provider) *Validator {
```

Comment:
Would it make sense to pass the registry here instead? And then do away with defaultConf and just use `reg.Config(ctx)`?

Reply:
Good idea